### PR TITLE
Use directly fugit instead of rufus-scheduler

### PR DESCRIPTION
**Pull request by jmettraux**
Addresses https://github.com/ondrejbartas/sidekiq-cron/issues/199

Sidekiq-cron is not using rufus-scheduler itself but rather its cron logic which recently got spun off in the fugit gem (https://github.com/floraison/fugit).

This patch makes Sidekiq-cron depend directly on fugit instead of rufus-scheduler.

A refinement would be to cache the parsed cron line in the Sidekiq-cron Job instance so that there is no need to reparse each time.

Here is the `rake test` output after patching. Please note that I did not address the remaining five errors, they seem unrelated, out of the scope for this patch :  [test1.txt](https://github.com/ondrejbartas/sidekiq-cron/files/2015048/test1.txt)

### CODE REVIEW

**Comment by jmettraux** [comment]
Good morning Nathan,

> The core of your change that swaps rufus-scheduler logic with fugit feels muddied by some changes you've included that are unrelated.

OK, I contribute mud.

> We were also recently surprised by the sudden loss of compatibility and would love to see sidekiq-cron move back to stability.

"sudden loss of compatibility"... Please read http://jmettraux.skepti.ch/20180601.html?t=Gemfile_lock&f=200

Regarding `!errors.any?` and the typo fix, granted, it's unrelated.

Regarding `sidekiq-cron.gemspec`, please read https://github.com/ondrejbartas/sidekiq-cron/blob/master/sidekiq-cron.gemspec#L1-L5

> I respect that you feel strongly about your chosen styles

That leaves just `!errors.any?` vs `errors.empty?` as part of my chosen styles.

I trust @ondrejbartas will do the right thing when he has time. I don't care if he rejects my PR, it's rather a message from me to him, showing how to go from [rufus-scheduler](https://github.com/jmettraux/rufus-scheduler) to [fugit](https://github.com/floraison/fugit). I have interacted with him before and I must say he's a nice chap. I'm sure he'll do what he, as project owner, feels is right and I will respect his choices.

Kind regards.

**Comment by Supernats** [comment]
I didn't see that line about using Jeweler. Thanks for pointing that out to me.

I believe I misspoke with the my use of the word "sudden," and for that I apologize. In hindsight, a better word would have been "unexpected," or perhaps no descriptor at all. We check in our Gemfile.lock, as you are absolutely correct in advising others do, and this never caused us a moment of downtime.

Thanks for your contributions and taking the time to courteously address someone looking to help this gem move back toward stability for the community.

**Comment by fschwahn** [comment]
The test failures are a result of a weird issue I think, namely that tests fail when run after 11pm because of https://github.com/ondrejbartas/sidekiq-cron/blob/40f179f6aa87c7d52034b3cf781b80929e5d72a8/test/integration/performance_test.rb#L32

If hour is `23`, it will create an invalid time.

What's interesting though is that one of the performance tests fails in all ruby versions. Is it possible that the update to `fugit` could cause such a performance regression? I looked at the history of test failures of the repo, and couldn't find another instance of it failing.

**Comment by jmettraux** [comment]
Caching the parsed cron instance solved the performance regression.

Green.

Thanks again @fschwahn

File: sidekiq-cron.gemspec [hunk 1]
```diff
@@ -37,7 +37,9 @@ Gem::Specification.new do |s|
     "lib/sidekiq/cron/launcher.rb",
     "lib/sidekiq/cron/locales/de.yml",
     "lib/sidekiq/cron/locales/en.yml",
+    "lib/sidekiq/cron/locales/ja.yml",
```
[the remark this answers is not in the document]
Comment (ondrejbartas) [reply sidekiq-cron.gemspec]: maybe this wasn't added because of publish

**Pull request merged**